### PR TITLE
Reconstruct Symmetric using uplo, and generalize for Hermitian also

### PR DESCRIPTION
**Pull request by sethaxen**
The `to_vec` function for `Hermitian` uses the same approach as used in #85.

### CODE REVIEW

**Comment by willtebbutt** [comment]
I guess we merge #85 then this. Please bump the patch version to 0.10.2

File: src/to_vec.jl [hunk 1]
```diff
@@ -55,11 +55,19 @@ end
 
 function to_vec(x::Symmetric)
     function Symmetric_from_vec(x_vec)
-        return Symmetric(reshape(x_vec, size(x)))
+        return Symmetric(reshape(x_vec, size(x)), Symbol(x.uplo))
```
Comment (nickrobinson251) [inline src/to_vec.jl:58]: Why do we need the Symbol conversion here (same below)?

Comment (sethaxen) [reply src/to_vec.jl:58]: Even though `uplo` is internally a char, the `Symmetric(...)` constructor is only defined for `uplo::Symbol`. The alternative to this approach is to use the constructor with types: `typeof(x)(..., x.uplo)`.

Comment (nickrobinson251) [reply src/to_vec.jl:58]: Oh, wild. That seems odd, but maybe there's a reason for it. Thanks for explaining 😊

Comment (sethaxen) [reply src/to_vec.jl]: I switched over to the type constructor.

Comment (sethaxen) [reply src/to_vec.jl]: Yeah, there's some strangeness around `Symmetric` and `Hermitian`. 🤷‍♂️ No problem!

**Comment by sethaxen** [comment]
Can this also be merged?

**Comment by sethaxen** [comment]
Thanks!

**Pull request merged**